### PR TITLE
js sample code for pronunciation assessment

### PR DESCRIPTION
**Pull request by l17813182030**
## Purpose

Provide JavaScript sample code for Pronunciation Assessment

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Comment by l17813182030** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by wangkenpu** [comment]
Approved.

File: samples/js/node/package.json [hunk 1]
```diff
@@ -10,10 +10,15 @@
   "author": "Microsoft",
   "license": "MIT",
   "dependencies": {
+    "difflib": "^0.2.4",
```
Comment (yulin-li) [inline samples/js/node/package.json:13]: check if this is in third party notice

File: samples/js/node/pronunciationAssessment.js [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        let resTextArray = [];
+        if (["zh-cn"].includes(settings.language.toLowerCase())) {
+            const resTextProcessed = (resText.toLocaleLowerCase() ?? "").replace(new RegExp("[^a-zA-Z0-9\u4E00-\u9FA5']+", "g"), " ");
+            const wholelyrics = (reference_text.toLocaleLowerCase() ?? "").replace(new RegExp("[^a-zA-Z0-9\u4E00-\u9FA5']+", "g"), " ");
+            const segment = new Segment();
+            segment.useDefault();
+            segment.loadDict('wildcard.txt');
+            _.map(segment.doSegment(wholelyrics, {stripPunctuation: true}), (res) => wholelyricsArry.push(res['w']));
+            _.map(segment.doSegment(resTextProcessed, {stripPunctuation: true}), (res) => resTextArray.push(res['w']));
+        } else {
```
Comment (yulin-li) [inline samples/js/node/pronunciationAssessment.js:105]: how about other non-English locales?

Comment (wangkenpu) [reply samples/js/node/pronunciationAssessment.js:105]: We just provide `zh-CN` word segmentation sample and for other locales, we just remove the punctuation like what we have done in the Python sample code.

Comment (yulin-li) [reply samples/js/node/pronunciationAssessment.js:105]: Does the service only support `zh-cn` and `en-us`? If not, please update readme to clearify the limitation of the sample codes

Comment (wangkenpu) [reply samples/js/node/pronunciationAssessment.js:105]: No. Support all locale. But for other locales which need word segmentation, the segmentation should be implemented by users. For other non-English locales, the non-word characters (e.g, punctuation) should be removed by users. We can not provide samples for all locales.
Anyway, @l17813182030 you can add some comments here to notice the users.

Comment (yulin-li) [reply samples/js/node/pronunciationAssessment.js:105]: Is there some library could be leveraged?

**Comment by l17813182030** [comment]
updated

File: samples/js/node/pronunciationAssessment.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+const filePushStream = require("./filePushStream");
+const _ = require('lodash') ;
+const forEach = require('lodash.foreach');
+const Segment = require('segment');
+var difflib = require('difflib');
+
+
+(function() {
+"use strict";
+
```
Comment (wangkenpu) [inline samples/js/node/pronunciationAssessment.js:15]: add the following comment

`pronunciation assessment with audio streaming and continue mode`

**Review by l17813182030 — COMMENTED** [review]
added

File: samples/js/node/pronunciationAssessment.js (hunk 3 above)
Comment (l17813182030) [reply samples/js/node/pronunciationAssessment.js:15]: added

Comment (wangkenpu) [reply samples/js/node/pronunciationAssessment.js:15]: Please also show the streaming progress in your code.

File: samples/js/node/pronunciationAssessment.js (hunk 2 above)
Comment (l17813182030) [reply samples/js/node/pronunciationAssessment.js:108]: > No. Support all locale. But for other locales which need word segmentation, the segmentation should be implemented by users. For other non-English locales, the non-word characters (e.g, punctuation) should be removed by users. We can not provide samples for all locales. Anyway, @l17813182030 you can add some comments here to notice the users.

have added comments to notice the users

File: samples/js/node/package.json (hunk 1 above)
Comment (l17813182030) [reply samples/js/node/package.json:13]: added

File: samples/js/node/pronunciationAssessment.js (hunk 3 above)
Comment (l17813182030) [reply samples/js/node/pronunciationAssessment.js:15]: update to display recognition process

File: samples/js/node/pronunciationAssessment.js (hunk 2 above)
Comment (l17813182030) [reply samples/js/node/pronunciationAssessment.js:111]: > Is there some library could be leveraged?

No similar library is found in js

**Review by yulin-li — DISMISSED** [review]

**Review by yulin-li — APPROVED** [review]

**Pull request merged**